src/accumulator/accumulated_map.rs

```diff
@@ -7,10 +9,6 @@ use super::{accumulated::Accumulated, Accumulator, AnyAccumulated};
 #[derive(Default, Debug)]
 pub struct AccumulatedMap {
     map: FxHashMap<IngredientIndex, Box<dyn AnyAccumulated>>,
```

Comment:
I guess a more local change would be to change `map` here to `Option<Box<FxHashMap>>`. It would have the advantage that all accumualted fields remain closer together.

Reply:
Generally not a bad idea but I think (will have to verify but I am fairly certain) that moving `InputAccumulatedValues` back into `AccumulatedMap` will increase the size of `QueryRevisions` as the field can't be batched together with other fields that are smaller than the alignment.

Reply:
I do think the field here should be option-boxed though instead, that should clean up the code interacting with `AccumulatedMap` at least